src/ip/QosConfig.cc

```diff
@@ -77,70 +77,81 @@ Ip::Qos::getTosFromServer(const Comm::ConnectionPointer &server, fde *clientFde)
 #endif
 }
 
-void Ip::Qos::getNfmarkFromServer(const Comm::ConnectionPointer &server, const fde *clientFde)
+nfmark_t
+Ip::Qos::getNfmarkFromConnection(const Comm::ConnectionPointer &conn, const Ip::Qos::ConnectionDirection connDir)
 {
+    nfmark_t mark = 0;
 #if USE_LIBNETFILTERCONNTRACK
     /* Allocate a new conntrack */
     if (struct nf_conntrack *ct = nfct_new()) {
 
         /* Prepare data needed to find the connection in the conntrack table.
          * We need the local and remote IP address, and the local and remote
          * port numbers.
+         * For connection from client we need swap remote with local, because
+         * src = client address(remote), dst = our address(local)
          */
+        auto dst = conn->remote;
+        auto src = conn->local;
```

Comment:
If possible, make these constant.

Please use one-line initialization and removing the comment about "swapping" as no longer necessary:

```
const auto src = (connDir == Ip::Qos::dirAccepted) ? conn->remote : conn->local;
const auto dst = (connDir == Ip::Qos::dirAccepted) ? conn->local : conn->remote;
```